### PR TITLE
save dictionary within checkpoint file

### PR DESCRIPTION
**Pull request by ramchandra-karandikar**
Save configuration dictionary within checkpoint Meta.
With this change, we don't need to worry about serializing configuration file to be saved within checkpoints

### CODE REVIEW

File: tools/train.py [hunk 1]
```diff
@@ -200,7 +200,7 @@ def main():
     logger.info('Environment info:\n' + dash_line + env_info + '\n' +
                 dash_line)
     meta['env_info'] = env_info
-    meta['config'] = cfg.pretty_text
+    meta['config'] = cfg.to_dict() #save configuration as a dictionary within checkpoint
```
Comment (siddharthanpr) [inline tools/train.py:203]: does this affect training? Can you change the mmdet32/apis/convert.py to read config from the model file itself?

**Review by siddharthanpr — COMMENTED** [review]
small comments

File: tools/train.py (hunk 1 above)
Comment (ramchandra-karandikar) [reply tools/train.py:203]: a. We need to open the model and embed configuration dictionary and save it again.
b. Testing out the training part: That will take a day or so

Comment (siddharthanpr) [reply tools/train.py:203]: Save the deterministic model file (optionally non deterministic). save the config in a new key. Maybe call it `export_config`

**Pull request closed without merging**